// react-row-select-table/dist/components/stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.__RewireAPI__ = exports.__ResetDependency__ = exports.__set__ = exports.__Rewire__ = exports.__GetDependency__ = exports.__get__ = undefined;

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _reactDom = require("react-dom");

var _reactDom2 = _interopRequireDefault(_reactDom);

var _react3 = require("@storybook/react");

var _addonKnobs = require("@storybook/addon-knobs");

var _addonActions = require("@storybook/addon-actions");

var _Table = require("./Table");

var _Table2 = _interopRequireDefault(_Table);

var _CustomSyncTable = require("./CustomSyncTable");

var _CustomSyncTable2 = _interopRequireDefault(_CustomSyncTable);

var _Thead = require("./Thead");

var _Thead2 = _interopRequireDefault(_Thead);

var _Tbody = require("./Tbody");

var _Tbody2 = _interopRequireDefault(_Tbody);

var _Tr = require("./Tr");

var _Tr2 = _interopRequireDefault(_Tr);

var _Th = require("./Th");

var _Th2 = _interopRequireDefault(_Th);

var _Td = require("./Td");

var _Td2 = _interopRequireDefault(_Td);

var _CustomTable = require("./CustomTable");

var _CustomTable2 = _interopRequireDefault(_CustomTable);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Test = function (_get__2) {
  _inherits(Test, _get__2);

  function Test() {
    _classCallCheck(this, Test);

    return _possibleConstructorReturn(this, (Test.__proto__ || Object.getPrototypeOf(Test)).apply(this, arguments));
  }

  _createClass(Test, [{
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps() {
      var root = _get__("ReactDOM").findDOMNode(this.root);
      if (!root) {
        return;
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      return _get__("React").createElement(
        "div",
        { ref: function ref(node) {
            return _this2.root = node;
          } },
        _get__("React").createElement(_get__("Custom"), null)
      );
    }
  }]);

  return Test;
}(_get__("Component"));

var Custom = function Custom() {
  return _get__("React").createElement(
    _get__("CustomSyncTable"),
    { onCheck: _get__("action")("onCheck"), checkeds: [1, 2, 3] },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(
      _get__("Tbody"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "1"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "tarou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "2"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "zirou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "3"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "subrou"
        )
      )
    )
  );
};

_get__("storiesOf")("DefaultTable", module).addDecorator(_get__("withKnobs")).add("Test", function () {
  return _get__("React").createElement(_get__("Test"), { on: _get__("boolean")("on", true) });
}).add("CustomTable", function () {
  return _get__("React").createElement(
    _get__("CustomTable"),
    { onCheck: _get__("action")("onCheck") },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(
      _get__("Tbody"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "1"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "tarou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "2"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "zirou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "3"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "subrou"
        )
      )
    )
  );
}).add("default checkeds", function () {
  return _get__("React").createElement(
    _get__("CustomTable"),
    { onCheck: _get__("action")("onCheck"), defaultCheckeds: [1, 2] },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(
      _get__("Tbody"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "1"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "tarou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "2"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "zirou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "3"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "subrou"
        )
      )
    )
  );
}).add("isCheckRow = false", function () {
  return _get__("React").createElement(
    _get__("CustomTable"),
    { onCheck: _get__("action")("onCheck") },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(
      _get__("Tbody"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        { isCheckRow: false },
        _get__("React").createElement(
          _get__("Td"),
          null,
          "1"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "tarou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        { isCheckRow: false },
        _get__("React").createElement(
          _get__("Td"),
          null,
          "2"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "zirou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        { isCheckRow: false },
        _get__("React").createElement(
          _get__("Td"),
          null,
          "3"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "subrou"
        )
      )
    )
  );
}).add("row 0", function () {
  return _get__("React").createElement(
    _get__("CustomTable"),
    { onCheck: _get__("action")("onCheck") },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(_get__("Tbody"), null)
  );
}).add("CustomSyncTable", function () {
  return _get__("React").createElement(
    _get__("CustomSyncTable"),
    {
      onCheck: _get__("action")("onCheck"),
      checkeds: _get__("array")("checked", [1, 2])
    },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(
      _get__("Tbody"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "1"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "tarou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "2"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "zirou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "3"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "subrou"
        )
      )
    )
  );
});
_get__("storiesOf")("BaseTable", module).addDecorator(_get__("withKnobs")).add("BaseTable", function () {
  return _get__("React").createElement(
    _get__("Table"),
    { onCheck: _get__("action")("onCheck") },
    _get__("React").createElement(
      _get__("Thead"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Th"),
          null,
          "id"
        ),
        _get__("React").createElement(
          _get__("Th"),
          null,
          "name"
        )
      )
    ),
    _get__("React").createElement(
      _get__("Tbody"),
      null,
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "1"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "tarou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "2"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "zirou"
        )
      ),
      _get__("React").createElement(
        _get__("Tr"),
        null,
        _get__("React").createElement(
          _get__("Td"),
          null,
          "3"
        ),
        _get__("React").createElement(
          _get__("Td"),
          null,
          "subrou"
        )
      )
    )
  );
});

function _getGlobalObject() {
  try {
    if (!!global) {
      return global;
    }
  } catch (e) {
    try {
      if (!!window) {
        return window;
      }
    } catch (e) {
      return this;
    }
  }
}

;
var _RewireModuleId__ = null;

function _getRewireModuleId__() {
  if (_RewireModuleId__ === null) {
    var globalVariable = _getGlobalObject();

    if (!globalVariable.__$$GLOBAL_REWIRE_NEXT_MODULE_ID__) {
      globalVariable.__$$GLOBAL_REWIRE_NEXT_MODULE_ID__ = 0;
    }

    _RewireModuleId__ = __$$GLOBAL_REWIRE_NEXT_MODULE_ID__++;
  }

  return _RewireModuleId__;
}

function _getRewireRegistry__() {
  var theGlobalVariable = _getGlobalObject();

  if (!theGlobalVariable.__$$GLOBAL_REWIRE_REGISTRY__) {
    theGlobalVariable.__$$GLOBAL_REWIRE_REGISTRY__ = Object.create(null);
  }

  return __$$GLOBAL_REWIRE_REGISTRY__;
}

function _getRewiredData__() {
  var moduleId = _getRewireModuleId__();

  var registry = _getRewireRegistry__();

  var rewireData = registry[moduleId];

  if (!rewireData) {
    registry[moduleId] = Object.create(null);
    rewireData = registry[moduleId];
  }

  return rewireData;
}

(function registerResetAll() {
  var theGlobalVariable = _getGlobalObject();

  if (!theGlobalVariable['__rewire_reset_all__']) {
    theGlobalVariable['__rewire_reset_all__'] = function () {
      theGlobalVariable.__$$GLOBAL_REWIRE_REGISTRY__ = Object.create(null);
    };
  }
})();

var INTENTIONAL_UNDEFINED = '__INTENTIONAL_UNDEFINED__';
var _RewireAPI__ = {};

(function () {
  function addPropertyToAPIObject(name, value) {
    Object.defineProperty(_RewireAPI__, name, {
      value: value,
      enumerable: false,
      configurable: true
    });
  }

  addPropertyToAPIObject('__get__', _get__);
  addPropertyToAPIObject('__GetDependency__', _get__);
  addPropertyToAPIObject('__Rewire__', _set__);
  addPropertyToAPIObject('__set__', _set__);
  addPropertyToAPIObject('__reset__', _reset__);
  addPropertyToAPIObject('__ResetDependency__', _reset__);
  addPropertyToAPIObject('__with__', _with__);
})();

function _get__(variableName) {
  var rewireData = _getRewiredData__();

  if (rewireData[variableName] === undefined) {
    return _get_original__(variableName);
  } else {
    var value = rewireData[variableName];

    if (value === INTENTIONAL_UNDEFINED) {
      return undefined;
    } else {
      return value;
    }
  }
}

function _get_original__(variableName) {
  switch (variableName) {
    case "ReactDOM":
      return _reactDom2.default;

    case "Component":
      return _react.Component;

    case "action":
      return _addonActions.action;

    case "storiesOf":
      return _react3.storiesOf;

    case "withKnobs":
      return _addonKnobs.withKnobs;

    case "boolean":
      return _addonKnobs.boolean;

    case "array":
      return _addonKnobs.array;

    case "React":
      return _react2.default;

    case "Custom":
      return Custom;

    case "CustomSyncTable":
      return _CustomSyncTable2.default;

    case "Thead":
      return _Thead2.default;

    case "Tr":
      return _Tr2.default;

    case "Th":
      return _Th2.default;

    case "Tbody":
      return _Tbody2.default;

    case "Td":
      return _Td2.default;

    case "Test":
      return Test;

    case "CustomTable":
      return _CustomTable2.default;

    case "Table":
      return _Table2.default;
  }

  return undefined;
}

function _assign__(variableName, value) {
  var rewireData = _getRewiredData__();

  if (rewireData[variableName] === undefined) {
    return _set_original__(variableName, value);
  } else {
    return rewireData[variableName] = value;
  }
}

function _set_original__(variableName, _value) {
  switch (variableName) {}

  return undefined;
}

function _update_operation__(operation, variableName, prefix) {
  var oldValue = _get__(variableName);

  var newValue = operation === '++' ? oldValue + 1 : oldValue - 1;

  _assign__(variableName, newValue);

  return prefix ? newValue : oldValue;
}

function _set__(variableName, value) {
  var rewireData = _getRewiredData__();

  if ((typeof variableName === "undefined" ? "undefined" : _typeof(variableName)) === 'object') {
    Object.keys(variableName).forEach(function (name) {
      rewireData[name] = variableName[name];
    });
  } else {
    if (value === undefined) {
      rewireData[variableName] = INTENTIONAL_UNDEFINED;
    } else {
      rewireData[variableName] = value;
    }

    return function () {
      _reset__(variableName);
    };
  }
}

function _reset__(variableName) {
  var rewireData = _getRewiredData__();

  delete rewireData[variableName];

  if (Object.keys(rewireData).length == 0) {
    delete _getRewireRegistry__()[_getRewireModuleId__];
  }

  ;
}

function _with__(object) {
  var rewireData = _getRewiredData__();

  var rewiredVariableNames = Object.keys(object);
  var previousValues = {};

  function reset() {
    rewiredVariableNames.forEach(function (variableName) {
      rewireData[variableName] = previousValues[variableName];
    });
  }

  return function (callback) {
    rewiredVariableNames.forEach(function (variableName) {
      previousValues[variableName] = rewireData[variableName];
      rewireData[variableName] = object[variableName];
    });
    var result = callback();

    if (!!result && typeof result.then == 'function') {
      result.then(reset).catch(reset);
    } else {
      reset();
    }

    return result;
  };
}

exports.__get__ = _get__;
exports.__GetDependency__ = _get__;
exports.__Rewire__ = _set__;
exports.__set__ = _set__;
exports.__ResetDependency__ = _reset__;
exports.__RewireAPI__ = _RewireAPI__;
exports.default = _RewireAPI__;
